### ISSUE TITLE
Bug with JSDoc comments for `multiline-comment-style` - `separate-lines` option

### ISSUE THREAD

**Issue by 0xCLARITY** [eslint/eslint#13405]
**Tell us about your environment**

```
Node version: v14.4.0
npm version: v6.14.4
Local ESLint version: v7.2.0 (Currently used)
Global ESLint version: Not found
```

**What parser (default, Babel-ESLint, etc.) are you using?**

`@typescript-eslint/parser: 3.2.0`

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  root: true,

  parser: '@typescript-eslint/parser', // Make ESLint compatible with TypeScript
  parserOptions: {
    // Enable linting rules with type information from our tsconfig
    tsconfigRootDir: __dirname,
    project: ['./tsconfig.eslint.json'],

    sourceType: 'module', // Allow the use of imports / ES modules

    ecmaFeatures: {
      impliedStrict: true, // Enable global strict mode
    },
  },

  // Specify global variables that are predefined
  env: {
    node: true, // Enable node global variables & Node.js scoping
    es2020: true, // Add all ECMAScript 2020 globals and automatically set the ecmaVersion parser option to ES2020
  },

  plugins: ['import', 'node'],

  extends: ['@xpring-eng/eslint-config-base/loose'],

  rules: {
    // We explicitly use `process.exit()` because all other errors should really be handled.
    'no-process-exit': 'off',
    'node/no-process-exit': 'off',

    // TODO:(hbergren) These are all rules we should remove eventually
    'import/max-dependencies': ['warn', { max: 9 }],
    'max-lines-per-function': ['warn', { max: 86 }],
    'max-statements': ['warn', { max: 26 }],
    complexity: ['warn', { max: 11 }],
  },

  // TODO:(hbergren) Should be able tor remove these overrides when we remove the above rules
  overrides: [
    {
      files: ['test/**/*.ts'],
      rules: {
        // Warn when modules have too many dependencies (code smell)
        // Increased the max for test files and test helper files, since tests usually need to import more things
        'import/max-dependencies': ['warn', { max: 8 }],

        // describe blocks count as a function in Mocha tests, and can be insanely long
        'max-lines-per-function': 'off',

        // TODO:(hbergren) We should probably cut this override eventually
        'max-lines': ['warn', {max: 500}],
      },
    },
  ],
}
```
</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```ts
/** This should be JSDoc
 * but ESLint flags it as saying I should be using // line comments
 */
const abc = 5
```

```bash
eslint . --ext .ts
```

**What did you expect to happen?**
Given my configuration of: `'multiline-comment-style': ['warn', 'separate-lines'],`

and that `multiline-comment-style`:
> ... always ignores directive comments such as /* eslint-disable */. Additionally, unless the mode is "starred-block", the rule ignores JSDoc comments.

I would expect that my multi-line JSDoc comment above a variable of `/** ... */` would not trigger a rule warning.

**What actually happened? Please include the actual, raw output from ESLint.**
`Expected multiple line comments instead of a block comment                                                 multiline-comment-style`.

**Are you willing to submit a pull request to fix this bug?**
Yes, if pointed in the right direction.
